test_contextual_lenses/tape_gfp.py

```diff
@@ -0,0 +1,673 @@
+
+  return test_batches
+
+test_fluorescences = test_df['log_fluorescence'].values
+
+
+"""Model evaluation."""
+def evaluate(optimizer, test_data, true_fluorescences, plot_title):
+
+  pred_fluorescences = []
+  for batch in iter(test_data):
+    X, Y = batch
+    preds = optimizer.target(X)
+    for pred in preds:
+      pred_fluorescences.append(pred[0])
+  pred_fluorescences = np.array(pred_fluorescences)
+  
+  spearmanr = scipy.stats.spearmanr(true_fluorescences, pred_fluorescences).correlation
+  mse = sklearn.metrics.mean_squared_error(true_fluorescences, pred_fluorescences)
+  print("Full data set:")
+  print("spearmanr: {}\nmse: {}".format(spearmanr, mse))
+  plt.scatter(true_fluorescences, pred_fluorescences)
+  plt.xlabel('True Fluorescences')
+  plt.ylabel('Predicted Fluorescences')
+  plt.title(plot_title)
+  plt.show()
+
+  bright_inds = np.where(true_fluorescences > 2.5)
+  bright_true_fluorescences = true_fluorescences[bright_inds]
+  bright_pred_fluorescences = pred_fluorescences[bright_inds]
+  bright_spearmanr = scipy.stats.spearmanr(bright_true_fluorescences, bright_pred_fluorescences).correlation
+  bright_mse = sklearn.metrics.mean_squared_error(bright_true_fluorescences, bright_pred_fluorescences)
+  print("Bright mode only:")
+  print("spearmanr: {}\nmse: {}".format(bright_spearmanr, bright_mse))
+  plt.scatter(bright_true_fluorescences, bright_pred_fluorescences)
+  plt.xlabel('True Fluorescences')
+  plt.ylabel('Predicted Fluorescences')
+  plt.title(plot_title + ' (Bright)')
+  plt.show()
+
+  dark_inds = np.where(true_fluorescences < 2.5)
+  dark_true_fluorescences = true_fluorescences[dark_inds]
+  dark_pred_fluorescences = pred_fluorescences[dark_inds]
+  dark_spearmanr = scipy.stats.spearmanr(dark_true_fluorescences, dark_pred_fluorescences).correlation
+  dark_mse = sklearn.metrics.mean_squared_error(dark_true_fluorescences, dark_pred_fluorescences)
+  print("Dark mode only:")
+  print("spearmanr: {}\nmse: {}".format(dark_spearmanr, dark_mse))
+  plt.scatter(dark_true_fluorescences, dark_pred_fluorescences)
+  plt.xlabel('True Fluorescences')
+  plt.ylabel('Predicted Fluorescences')
+  plt.title(plot_title + ' (Dark)')
+  plt.show()
+
+  results = {
+      'spearmanr': spearmanr,
+      'mse': mse,
+      'pred_fluorescences': pred_fluorescences,
+      'bright_spearmanr': bright_spearmanr,
+      'bright_mse': bright_mse,
+      'bright_pred_fluorescences': bright_pred_fluorescences,
+      'dark_spearmanr': dark_spearmanr,
+      'dark_mse': dark_mse,
+      'dark_pred_fluorescences': dark_pred_fluorescences
+  }
+
+  return results
+
+"""Experiments."""
+
+"""Mean pool lens."""
+epochs = 50
+train_batches = create_train_data(epochs, train_bs=256)
+test_batches = create_test_data(test_bs=256)
+
+lr = 1e-1
+wd = 0.
+
+encoder_fn = one_hot_encoder
+encoder_fn_kwargs = {
+    
+}
+reduce_fn = mean_pool
+reduce_fn_kwargs = {
+    
+}
+
+mean_pool_model = create_representation_model(encoder_fn=encoder_fn,
```

Comment:
Aren't the GFP sequences all the same length? You should definitely benchmark against a reduce_fn that just flattens the one-hots.

For that, it would be good to check the difference in performance from your linear model on top vs. a sklearn linear regression model. They should get the same performance.